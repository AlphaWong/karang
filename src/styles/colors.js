export const orange = 'rgba(242,103,34,1)';
export const gold = 'rgba(250,162,27,1)';
export const red = 'rgba(216,83,79,1)';
export const green = 'rgba(0,137,91,1)';
export const white = 'rgba(255,255,255,1)';
export const offWhite = 'rgba(229,229,229,1)';
export const silver = 'rgba(128,130,133,1)';
export const gray = 'rgba(88,89,91,1)';
export const black = 'rgba(51,49,50,1)';
export const facebook = 'rgba(59, 89, 152, 1)';
export const lightGray = 'rgba(180, 180, 180, 1)';
export const veryLightGray = 'rgba(242,242,242, 1)';

const colors = {
  orange,
  gold,
  white,
  red,
  green,
  gray,
  silver,
  offWhite,
  black,
  facebook,
  lightGray,
  veryLightGray,
};

export default colors;
