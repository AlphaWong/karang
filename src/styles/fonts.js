import { injectGlobal } from 'styled-components';

injectGlobal`
  @import url(https://fonts.googleapis.com/css?family=Noto+Sans:400,700);
  @import url(https://fonts.googleapis.com/earlyaccess/notosansjapanese.css);
  @import url(https://fonts.googleapis.com/earlyaccess/notosanstc.css);
`;

export const primaryFonts =
  'Noto Sans, "Noto Sans TC", "Noto Sans JP", sans-serif';

export const fontWeight = {
  normal: 400,
  bold: 700,
};

export default {
  primaryFonts,
  fontWeight,
};
