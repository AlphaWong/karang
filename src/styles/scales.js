export const baseFontSize = 14;

export default {
  baseFontSize,
};
