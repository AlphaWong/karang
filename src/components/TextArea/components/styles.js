import styled from 'styled-components';

import { orange, lightGray, red, offWhite } from 'styles/colors';
import { primaryFonts, fontSize } from 'styles/fonts';

const TextAreaContainer = styled.div`
  display: flex;
  flex-direction: column;
  box-sizing: border-box;
  height: 100%;
  width: 100%;
  padding: 11px 11px 6px 11px;
  border: 1px solid ${offWhite};
  ${({ error }) =>
    error
      ? `
    border: 1px solid ${red};
    `
      : `
    &:focus-within {
      border: 1px solid ${orange};
    }
    `};
`;

const CharacterLimitDisplay = styled.div`
  font-size: ${fontSize.small};
  font-family: ${primaryFonts};
  line-height: 16px;
  color: ${lightGray};
`;

const placeholderStyle = `
  color: ${lightGray};
  font-size: ${fontSize.regular};
  font-family: ${primaryFonts};
`;

const InnerInputArea = styled.textarea`
  display: block;
  flex-grow: 1;
  font-family: ${primaryFonts};
  width: 100%;
  border: 0px solid transparent;
  padding: 0px;
  resize: none;
  &:focus {
    outline-width: 0;
  }

  &::-webkit-input-placeholder {
    /* Chrome/Opera/Safari */
    ${placeholderStyle};
  }

  &::-moz-placeholder {
    /* Firefox 19+ */
    ${placeholderStyle};
  }

  &:-ms-input-placeholder {
    /* IE 10+ */
    ${placeholderStyle};
  }

  &:-moz-placeholder {
    /* Firefox 18- */
    ${placeholderStyle};
  }
`;

export { TextAreaContainer, CharacterLimitDisplay, InnerInputArea };
