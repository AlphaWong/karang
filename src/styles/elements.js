import { injectGlobal } from 'styled-components';
import { orange } from './colors';
import { baseFontSize } from './scales';
import { primaryFonts } from './fonts';

injectGlobal`
  html, body {
    height: 100%;
    font-family: ${primaryFonts};
    font-size: ${baseFontSize}px;
    line-height: 1.143;
  }

  a {
    color: ${orange};
    text-decoration: none;

    &:hover,
    &:focus {
      /* text-decoration: underline; */
    }

    &:active {
      text-decoration: none;
    }
  }
`;
