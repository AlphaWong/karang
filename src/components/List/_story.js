import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import InfoIcon from 'icons/Info';
import { orange } from 'styles/colors';

import List from './index';

storiesOf('List', module).add(
  'Basic',
  withInfo({})(() => (
    <List
      hoverable
      unique="id"
      items={[
        {
          id: 'aaa',
          name: 'Silom Complex',
          address: 'Si Lom Bang Rak Bangkok Thailand',
          contact: 'Kevin 0938278268',
        },
        {
          id: 'bbb',
          name: 'Times Square',
          address: 'Matheson Street, Causeway Bay, Hong Kong',
          contact: 'Kevin 0938278268',
        },
        {
          id: 'ccc',
          name: 'LANE CRAWFORD TIMES SQUARE',
          address: 'Causeway Bay, Hong Kong',
          contact: 'Kevin 0938278268',
        },
      ]}
    >
      {({ data: { name, address, contact }, Item, getProps }) => (
        <Item {...getProps()} icon={<InfoIcon color={orange} />}>
          <div>
            {name} <span className="light-silver">{address}</span>
          </div>
          <span className="f6 moon-gray">{contact}</span>
        </Item>
      )}
    </List>
  ))
);
