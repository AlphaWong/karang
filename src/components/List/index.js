import React from 'react';
import styled, { css } from 'styled-components';
import { lighten } from 'polished';
import { orange, offWhite, white } from 'styles/colors';
import { arrayOf, bool, func, node, string, shape } from 'prop-types';

import noop from 'utils/noop';

const resetList = css`
  list-style: none;
  padding: 0;
  margin: 0;
`;

const Content = styled.div`
  flex: 1;
  line-height: 20px;

  ${({ variant }) => {
    switch (variant) {
      case 'small':
        return css`
          padding: 6px 0 6px 0;
        `;
      case 'large':
      default:
        return css`
          padding: 12px 0 12px 0;
        `;
    }
  }};
`;

const UL = styled.ul`
  ${resetList} box-shadow: 0 1px 4px rgba(0, 0, 0, 0.22);
  display: inline-block;
  box-sizing: border-box;
  ${({ variant }) => {
    switch (variant) {
      case 'small':
        return css`
          min-width: 12.5rem;
        `;
      case 'large':
      default:
        return css`
          min-width: 16rem;
        `;
    }
  }};
`;

const activeStyle = css`
  background-color: ${lighten(0.05, offWhite)};
  border-left-color: ${orange};
`;

const LI = styled.li`
  padding-left: 8px;
  border-left: 2px solid ${white};
  background-color: ${white};
  outline: 0;

  ${({ variant }) => {
    switch (variant) {
      case 'small':
        return css`
          padding-right: 6px;
        `;
      case 'large':
      default:
        return css`
          padding-right: 20px;
        `;
    }
  }};

  &:not(:last-child) ${/* sc-selector */ Content} {
    ${({ variant }) => {
      switch (variant) {
        case 'small':
          return css`
            border-bottom: 0;
          `;
        case 'large':
        default:
          return css`
            border-bottom: 1px solid ${offWhite};
          `;
      }
    }};
  }

  ${({ hoverable }) =>
    hoverable &&
    css`
      &:hover,
      &:focus {
        ${activeStyle};
      }
    `};
  ${({ active }) => active && activeStyle};
`;

const Wrapper = LI.extend`
  display: flex;
  align-items: top;
  cursor: pointer;
`;

const Icon = styled.div`
  ${({ variant }) => {
    switch (variant) {
      case 'small':
        return css`
          margin: 7.5px 6px 0 0;
        `;
      case 'large':
      default:
        return css`
          margin-top: 12px;
          margin-right: 10px;
        `;
    }
  }};
`;

export const Item = ({ icon, variant, children, options, ...rest }) => (
  <Wrapper variant={variant} {...rest}>
    {icon && <Icon variant={variant}>{icon}</Icon>}
    <Content variant={variant}>{children}</Content>
    {options}
  </Wrapper>
);

const List = ({ children, items, hoverable, unique, variant, ...rest }) => (
  <UL variant={variant} {...rest}>
    {items.map((data, index) =>
      children({
        data,
        Item,
        index,
        getProps: props => ({
          ...props,
          hoverable,
          variant,
          key: unique ? data[unique] : index,
          tabIndex: 0,
        }),
      })
    )}
  </UL>
);

Item.defaultProps = {
  icon: null,
  variant: null,
  children: null,
  options: null,
};

Item.propTypes = {
  icon: node,
  variant: string,
  children: node,
  options: node,
};

List.defaultProps = {
  hoverable: false,
  unique: '',
  variant: null,
  children: noop,
};

List.propTypes = {
  unique: string,
  items: arrayOf(shape({})).isRequired,
  hoverable: bool,
  variant: string,
  children: func,
};

export default List;
