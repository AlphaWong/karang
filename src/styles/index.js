import './colors';
import './elements';
import './fonts';
import './generic';
import './scales';
