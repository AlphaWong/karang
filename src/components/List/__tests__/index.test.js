/* global render, mount */
import React from 'react';
import List from '../index';
import { lighten } from 'polished';
import { orange, offWhite } from 'styles/colors';

describe('List component', () => {
  it('default', () => {
    expect(
      render(
        <List items={[{}, {}, {}, {}]}>
          {({ Item, getProps }) => <Item {...getProps()}>hello</Item>}
        </List>
      )
    ).toMatchSnapshot();
  });

  it('with icon', () => {
    expect(
      render(
        <List items={[{}, {}, {}, {}]}>
          {({ Item, getProps }) => <Item {...getProps({ icon: 'blah' })} />}
        </List>
      )
    ).toMatchSnapshot();
  });

  it('use unique as key', () => {
    const data = [{ id: 'a' }, { id: 'b' }, { id: 'c' }, { id: 'd' }];
    const comp = mount(
      <List unique="id" items={data}>
        {({ Item, getProps }) => <Item {...getProps()} />}
      </List>
    );

    comp.find('LI').forEach((wrapper, i) => {
      expect(wrapper.key()).toEqual(data[i].id);
    });
  });

  it('hoverable', () => {
    const data = [{ id: 'a' }, { id: 'b' }, { id: 'c' }, { id: 'd' }];
    const comp = mount(
      <List hoverable items={data}>
        {({ Item, getProps }) => <Item {...getProps({ icon: 'blah' })} />}
      </List>
    );

    const listItem = comp.find('li').at(2); // randomly pick one

    expect(listItem).toHaveStyleRule('border-left-color', orange, {
      modifier: ':hover',
    });

    expect(listItem).toHaveStyleRule(
      'background-color',
      lighten(0.05, offWhite),
      {
        modifier: ':hover',
      }
    );
  });
});
