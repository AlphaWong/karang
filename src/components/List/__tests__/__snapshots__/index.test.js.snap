// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List component default 1`] = `
.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 20px;
  padding: 12px 20px 12px 0;
}

.c0 {
  list-style: none;
  padding: 0;
  margin: 0;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.22);
  display: inline-block;
  box-sizing: border-box;
  background-color: rgba(255,255,255,1);
  color: rgba(51,49,50,1);
  min-width: 16rem;
}

.c1 {
  position: relative;
  padding-left: 8px;
  border-left: 2px solid transparent;
  outline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: top;
  -webkit-box-align: top;
  -ms-flex-align: top;
  align-items: top;
  cursor: pointer;
}

.c1:not(:last-child) .c2 {
  border-bottom: 1px solid rgba(229,229,229,1);
}

<ul
  class="c0"
>
  <li
    class="c1"
    tabindex="0"
  >
    <div
      class="c2 c3"
    >
      hello
    </div>
  </li>
  <li
    class="c1"
    tabindex="0"
  >
    <div
      class="c2 c3"
    >
      hello
    </div>
  </li>
  <li
    class="c1"
    tabindex="0"
  >
    <div
      class="c2 c3"
    >
      hello
    </div>
  </li>
  <li
    class="c1"
    tabindex="0"
  >
    <div
      class="c2 c3"
    >
      hello
    </div>
  </li>
</ul>
`;

exports[`List component with icon 1`] = `
.c4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  font-family: "Noto Sans",sans-serif;
  font-size: 14px;
  line-height: 20px;
  padding: 12px 20px 12px 0;
}

.c0 {
  list-style: none;
  padding: 0;
  margin: 0;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.22);
  display: inline-block;
  box-sizing: border-box;
  background-color: rgba(255,255,255,1);
  color: rgba(51,49,50,1);
  min-width: 16rem;
}

.c1 {
  position: relative;
  padding-left: 8px;
  border-left: 2px solid transparent;
  outline: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: top;
  -webkit-box-align: top;
  -ms-flex-align: top;
  align-items: top;
  cursor: pointer;
}

.c1:not(:last-child) .c3 {
  border-bottom: 1px solid rgba(229,229,229,1);
}

.c2 {
  margin-top: 12px;
  margin-right: 10px;
}

<ul
  class="c0"
>
  <li
    class="c1"
    tabindex="0"
  >
    <div
      class="c2"
    >
      blah
    </div>
    <div
      class="c3 c4"
    >
      
    </div>
  </li>
  <li
    class="c1"
    tabindex="0"
  >
    <div
      class="c2"
    >
      blah
    </div>
    <div
      class="c3 c4"
    >
      
    </div>
  </li>
  <li
    class="c1"
    tabindex="0"
  >
    <div
      class="c2"
    >
      blah
    </div>
    <div
      class="c3 c4"
    >
      
    </div>
  </li>
  <li
    class="c1"
    tabindex="0"
  >
    <div
      class="c2"
    >
      blah
    </div>
    <div
      class="c3 c4"
    >
      
    </div>
  </li>
</ul>
`;
